Actualizar eventos meetup bot

Scot3004 opened the pull request:
Actualización de eventos sabados

![image](https://github.com/ColombiaPython/sitio-web/assets/1175402/27a94464-e6d8-4d76-b687-44690a1ac4c7)

https://github.com/Scot3004/sitio-web/pull/1

https://github.com/Scot3004/sitio-web/actions/runs/9472096159

Scot3004 committed:
Actualizar eventos meetup bot

pyjavo commented on .github/workflows/update-events.yml:
```diff
@@ -0,0 +1,63 @@
+
+on:
+  schedule:
+    - cron: "0 12 * * 6" # Todos los sabados a las 12:00
```
¿A qué se debe que sea cada sábado? Hasta el momento solo hay 2 comunidades activas y cada una tiene eventos mensuales

pyjavo replied:
@Scot3004 sigo a la espera de esta pregunta

pyjavo replied:
Propongo que sea mensual y que sea entre semana para no recibir notificaciones los fines de semana jajaja. Puede ser el primer miercoles del mes o algo asi @Scot3004

Scot3004 replied:
No me aparecia el comentario.
En cuanto al por que semanal los sabados es un efecto de copiar y pegar, en el de pybaq lo tenemos semanal los martes a las 12 😄  ten en cuenta que los eventos deberian actualizarse en 2 momentos, cuando se crea en meetup y cuando ya paso, entonces ese mismo evento en cada una de las 4 semanas apareceria de forma distinta

Scot3004 replied:
Sera correcta esta expresión?
0 0 1,2,3,4,5,6,7 * 3

![image](https://github.com/ColombiaPython/sitio-web/assets/1175402/1fc221d3-670a-467d-b2e8-0c71cfd12040)

pyjavo replied:
Esta arroja una mas simple ¿qué opinas?
https://crontab.guru/#5_4_15_*_*

Scot3004 replied:
Estoy dudando de la hora 4:05 siendo utc-5 seria a las 11pm que estaria ejecutandose

Scot3004 replied:
por otro lado se ejecutaria todos los 15 sin tener en cuenta si es domingo o festivo

Scot3004 replied:
![image](https://github.com/ColombiaPython/sitio-web/assets/1175402/ce5a921d-2fec-4a00-ac9b-160085dc6518)
Acabo de ver que la pagina que pasaste da cuando seria la proxima ejecución y me suena mas el 6 a las 7pm :D

Scot3004 replied:
acabo de caer en cuenta que no correria miercoles sino martes :)

pyjavo replied:
En ese caso si tocaria buscar una regla mas específica

Scot3004 replied:
Que tal esta, el primer miercoles del mes a medio dia?

![image](https://github.com/ColombiaPython/sitio-web/assets/1175402/3b19d08f-d58e-4b4a-bcdb-519fd26e9673)

pyjavo replied:
Lo que no me cuadra es que diga que que se ejecuta en 7 días diferentes y además el miércoles (1 and 2 and 3 and 4... and Wednesday).

Scot3004 committed:
Adicionado a blacklist evento roto

Al intentar abrirlo, aparece un error 500 de parte de meetup

Ademas la información del mismo se enecuentra incompleta

https://www.meetup.com/pythonbogota/events/238435129/

Scot3004 commented:
Adicione a blacklist el evento https://www.meetup.com/pythonbogota/events/238435129/ por que estaba roto, y no se deberia tener data invalida en el pr

https://github.com/Scot3004/sitio-web/pull/3
https://github.com/Scot3004/sitio-web/actions/runs/9605078422

Scot3004 committed:
Cambiado schedule a ejecutarse el primer del mes a mediodia

pyjavo commented on .github/workflows/update-events.yml:
```diff
@@ -2,7 +2,7 @@ name: "Actualizar eventos meetup"
 on:
   schedule:
-    - cron: "0 12 * * 6" # Todos los sabados a las 12:00
+    - cron: "0 17 1,2,3,4,5,6,7 * 3" # El primer miercoles del mes a mediodia UTC-5 Github esta en formato UTC
```
Aquí dice que se va a ejecutar 8 días incluyendo el miércoles (el 1ero de cada mes AND el 2do de cada mes AND el 3ero de cada mes AND .... AND miércoles)

Scot3004 replied:
En realidad es uno de esos 7, el que caiga miércoles, en el ejemplo de la imagen sería el 6, 1 y 8 caen el mismo día de semana, por eso solo coloque hasta el 7

arendondiosa commented on .github/workflows/update-events.yml:
```diff
@@ -0,0 +1,63 @@
+name: "Actualizar eventos meetup"
+
+on:
```
@Scot3004 puedes probar con este cambio para ejecutar el job antes de mezclarlo?
O lo probaste en algun otro repositorio propio?

```suggestion
on:
  pull_request:
    branches:
      - develop
```

Scot3004 replied:
No lo quise poner, la idea es que la actualización de eventos se haga bajo una de estas 2 premisas
* Una vez al mes para actualizar según lo que este en meetup, para que no se nos quede tan atras como ahora que se queda un año y nadie envia un pr actualizando eventos
* A demanda, por ejemplo si quiero promocionar un evento en mi grupo local y lo acabo de crear, con este dispatch puedo forzar que se actualicen los eventos

En cuanto al por que no considero conveniente que se corra cuando se hace un pull request, cada vez que se haga un pull request se creara otro con los eventos actualizados, a decir verdad para eso tenemos `deploy.yml`  

En si este job solo sirve para actualizar los eventos es algo así como una etl que extrae la data de meetup, la transforma en content.lr y luego lo carga en github

Scot3004 replied:
Aca está el ejemplo de como deberia funcionar

https://github.com/PyBAQ/website/actions/workflows/actualizar-eventos-meetup.yml

Ten en cuenta que tiene ciertas diferencias, pero en esencia hacen lo mismo
